### ISSUE TITLE
Error synchronizing products (Swift.DecodingError) on regular_price

### ISSUE THREAD

**Issue by reginabally** [woocommerce/woocommerce-ios#4737]
**Describe the bug**
Similar to [this issue](https://github.com/woocommerce/woocommerce-ios/issues/3511), the app wasn't able to load the products in the app because of the following error:

```
2021/08/05 17:05:26:053  ⛔️ Error loading ProductVariation & Product for siteID: 191752495 productID:27454 variationID:27459 error:AnyError(error: Swift.DecodingError.typeMismatch(Swift.String, Swift.DecodingError.Context(codingPath: [CodingKeys(stringValue: "data", intValue: nil), CodingKeys(stringValue: "regular_price", intValue: nil)], debugDescription: "Expected to decode String but found a number instead.", underlyingError: nil)))
```

**Steps to reproduce in SSP**
1. Go to Orders and select an order with product variations
2. Tap the variable product to view the product details.
3. The app shows an error "This product couldn't be loaded".
4. Go to Products and select the same product to view the product details.
5. Tap "Variations" and see that none of the product variation loads.

A screen recording of the SSP test was attached in 4201936-zd-woothemes.

**Mobile Environment**
Please include:
- Device: Reproduced on iPhone SE 2nd-gen
- iOS version: Reproduced on iOS 14.7.1
- WooCommerce iOS version: 7.1.0.2

**List of active plugins** (Complete SSR included in the ticket)
<details>
Redux: by Redux.io – 4.2.11
Advanced Product Fields for WooCommerce: by StudioWombat – 1.3.9
Akismet Anti-Spam: by Automattic – 4.1.10
Contact Form 7: by Takayuki Miyoshi – 5.4.2
Creative Mail by Constant Contact: by Constant Contact – 1.4.2
Duplicate Menu: by Jonathan Christopher – 0.2.2
Elementor Pro: by Elementor.com – 3.3.5
Elementor: by Elementor.com – 3.3.1
Email Template Customizer for WooCommerce: by VillaTheme – 1.0.4.4
Facebook Chat Plugin - Live Chat Plugin for WordPress: by Facebook – 2.2
Google Analytics for WordPress by MonsterInsights: by MonsterInsights – 7.18.0
Granular Controls Elementor: by Zulfikar Nore – 1.0.5
Header Footer Code Manager: by 99robots – 1.1.10
Elementor Header & Footer Builder: by Brainstorm Force
Nikhil Chavan – 1.6.3

S2W - Import Shopify to WooCommerce: by VillaTheme – 1.1.3.7
Jetpack: by Automattic – 10.0
MC4WP: Mailchimp for WordPress: by ibericode – 4.8.6
myStickymenu: by Premio – 2.5.6
NotificationX: by WPDeveloper – 1.9.4
Omnisend for Woocommerce: by Omnisend – 1.7.6
PixelYourSite: by PixelYourSite – 8.2.6
Rearrange Woocommerce Products: by Aslam Doctor – 2.3.6
Slider Revolution: by ThemePunch – 6.4.6
SiteGround Optimizer: by SiteGround – 5.8.3
Side Cart WooCommerce: by XootiX – 2.1
SVG Support: by Benbodhi – 2.3.19
TranslatePress - Multilingual: by Cozmoslabs
Razvan Mocanu
Madalin Ungureanu
Cristophor Hurduban – 2.0.6

Woo Discount Rules: by Flycart – 2.3.8
Product Slider and Carousel with Category for WooCommerce: by WP OnlineSupport – 2.5.2
Variation Swatches for WooCommerce: by Emran Ahmed – 1.1.17
BEAR - Bulk Editor and Products Manager: by realmag777 – 2.0.8.1
WooCommerce Stripe Gateway: by WooCommerce – 5.3.0
WooCommerce Payments: by Automattic – 2.8.2
WooCommerce PayPal Payments: by WooCommerce – 1.4.0
WooCommerce: by Automattic – 5.5.2
WordPress Importer: by wordpressdotorg – 0.7
Yoast SEO: by Team Yoast – 16.8
WordPress Starter: by SiteGround – 1.1.7
WP Crontrol: by John Blackbourn & crontributors – 1.10.0
WPForms Lite: by WPForms – 1.6.8.1
YITH WooCommerce Compare: by YITH – 2.5.3
YITH WooCommerce Quick View: by YITH – 1.6.3
YITH WooCommerce Wishlist: by YITH – 3.0.25
</details>

**Comment by iamgabrielma** [comment]
Some debugging:

The problem here seems to be that the JSON decoder expects the price to come as string, however it comes as an integer so it cannot be "translated" for the app to output this correctly on-screen.

This is due to a 3d party plugin altering both the "display_price", and "regular_price", if we check the `data source` in the inspector we'll see how there's a bunch of stuff going on, but specially the price is returning something like 20 decimal numbers:

<img width="1345" alt="Screenshot 2021-08-06 at 16 45 44" src="https://user-images.githubusercontent.com/3812076/128492296-93780e4b-4ee2-4541-a8fd-0eee110fe2e1.png">

**Comment by iamgabrielma** [comment]
Another case here: 4251719-zen

```
### Active Plugins (18) ###

ManageWP - Worker: by GoDaddy – 4.9.9
Advanced Custom Fields: by Delicious Brains – 5.10.1
Advanced Dynamic Pricing for WooCommerce: by AlgolPlus – 3.2.6
Elementor Pro: by Elementor.com – 3.3.7
Elementor: by Elementor.com – 3.4.2
Envato Elements: by Envato – 2.0.9
Facebook for WooCommerce: by Facebook – 2.6.1
Facebook Chat Plugin - Live Chat Plugin for WordPress: by Facebook – 2.2
Jetpack: by Automattic – 10.0
Google Ads & Marketing by Kliken: by Kliken – 1.0.7
Woocommerce Table Rate Shipping: by JEM Plugins – 2.1.1
WooCommerce Gift Cards: by SomewhereWarm – 1.8.0
WooCommerce PayPal Payments: by WooCommerce – 1.5.1
WooCommerce Shipping & Tax: by WooCommerce – 1.25.18
WooCommerce Square: by WooCommerce – 2.5.3
WooCommerce: by Automattic – 5.6.0
Smush Pro: by WPMU DEV – 3.8.8
WPMU DEV Dashboard: by WPMU DEV – 4.11.3
```

**Comment by reginabally** [comment]
Another report in 4261887-zd-woothemes.

```
### Active Plugins (10) ###

Ultimate Addons for WPBakery Page Builder: by Brainstorm Force – 3.19.11
Facebook for WooCommerce: by Facebook – 2.6.3-dev
Imagify: by WP Media – 1.10
Jetpack: by Automattic – 10.0
The7 WPBakery Page Builder: by Michael M - WPBakery.com – 6.6.0.1
UpdraftPlus - Backup/Restore: by UpdraftPlus.Com
DavidAnderson – 2.16.53.25

WooCommerce handepay: by Handepay – 1.3
WooCommerce Payments: by Automattic – 2.9.0
WooCommerce: by Automattic – 5.6.0
YITH WooCommerce Cart Messages: by YITH – 1.6.7
```

**Comment by reginabally** [comment]
@itsmeichigo We have another report in 4269667-zd-woothemes but I don't see WooCommerce Payments in the active plugin list. Will the PR fix the issue caused by a different plugin?

<details>
  <summary>Click to view list of active plugins</summary>
<pre>
StoreCustomizer: par Kaira – 2.3.2
ACF Front End Editor: par Horion Digital – 2.0.2
ACF to REST API: par Aires Gonçalves – 3.3.2
AppMySite: par AppMySite – 3.0.1
CMP - Coming Soon & Maintenance Plugin: par NiteoThemes – 4.0.15
WooCommerce Coupon Generator: par Jeroen Sormani – 1.2.0
Currency Switcher for WooCommerce Pro: par WP Wham – 2.14.0
Customer Reviews for WooCommerce: par CusRev – 4.23
Elementor Pro: par Elementor.com – 3.3.0
Elementor: par Elementor.com – 3.3.1
ElementsKit Lite: par Wpmet – 2.3.3
Facebook for WooCommerce: par Facebook – 2.6.1
Elementor Header & Footer Builder: par Brainstorm Force
Nikhil Chavan – 1.6.3

HT Menu Lite: par HasThemes – 1.1.6
Jetpack: par Automattic – 10.0
JetWoo Widgets For Elementor: par Crocoblock – 1.1.2
Google Ads & Marketing par Kliken: par Kliken – 1.0.7
Newsletter, SMTP, Email marketing and Subscribe forms by Sendinblue: par Sendinblue – 3.1.16
MailPoet 3 (nouveau): par MailPoet – 3.65.1
Make Column Clickable Elementor: par Fernando Acosta – 1.3.1
SG Woocommerce Delivery Location Map Picker: par Sevengits – 1.0.8
Menu Icons: par ThemeIsle – 0.12.8
Order Tracking - WordPress Status Tracking Plugin: par Etoile Web Design – 3.0.9
PI WooCommerce order date time and type PRO: par PI Websolution – 3.3.6.7
Product Size Chart for WooCommerce: par VillaTheme – 1.0.6
PW WooCommerce Gift Cards Pro: par Pimwick
LLC – 1.313

PW WooCommerce BOGO Free: par Pimwick
LLC – 2.33

QuadMenu: par Mega Menu – 2.1.2
Slider Revolution: par ThemePunch – 6.4.11
SiteGround Optimizer: par SiteGround – 5.9.6
Shortcode in Menus: par Gagan Deep Singh – 3.5.1
SVG Support: par Benbodhi – 2.3.19
Temporaire Login Without Password: par StoreApps – 1.6.14
TranslatePress - Multilingual: par Cozmoslabs
Razvan Mocanu
Madalin Ungureanu
Cristophor Hurduban – 2.1.0

Ultimate WP Mail: par Etoile Web Design – 1.0.6
WooCommerce Product Filter by WooBeWoo: par WooBeWoo – 1.5.7
WooCommerce Product Table by WooBeWoo: par WooBeWoo – 1.4.7
Woo Products Widgets For Elementor: par Themelocation – 1.0.5
Related Products for WooCommerce: par PeachPay – 3.3.7
WPC Smart Compare for WooCommerce: par WPClever – 4.0.5
WPC Smart Quick View for WooCommerce: par WPClever – 2.6.9
WPC Smart Wishlist for WooCommerce: par WPClever – 2.8.4
Variation Swatches for WooCommerce: par Emran Ahmed – 1.1.17
WooCommerce Admin: par WooCommerce – 2.6.1
WooCommerce Checkout Field Editor: par Woocommerce – 1.6.2
WooCommerce Colors: par WooThemes – 1.0.7
DPO Group plugin for WooCommerce: par DPO Group – 1.1.0
WooCommerce Shipping & Tax: par WooCommerce – 1.25.18
WooCommerce Table Rate Shipping: par WooCommerce – 3.0.32
WooCommerce: par Automattic – 5.6.0
WooStore Pro Api - by Aniket Malik (aniketmalikwork@gmail.com): par Aniket Malik – 2.1.1
Hustle: par WPMU DEV – 7.4.4
Yoast SEO: par L’équipe Yoast – 17.0
WordPress Starter: par SiteGround – 1.1.7
WP HTML Mail - Email Template Designer: par Hannes Etzelstorfer // codemiq – 3.0.9
WP Super Cache: par Automattic – 1.7.4
</pre>
</details>

**Comment by itsmeichigo** [comment]
@reginabally I don't think that is the same problem since the parsing failure was in product and not product variation. It's supposed to have been fixed by #3679 (which is 5 months ago). I'll see what I can find about this, but I think we'll need a separate issue for it.

Interestingly I've also found in that ticket's log an error parsing product variation's `permalink` where a number is found instead of a string:

> ⛔️ Error synchronizing product variations: typeMismatch(Swift.String, Swift.DecodingError.Context(codingPath: [CodingKeys(stringValue: "data", intValue: nil), _JSONKey(stringValue: "Index 0", intValue: 0), CodingKeys(stringValue: "permalink", intValue: nil)], debugDescription: "Expected to decode String but found a number instead.", underlyingError: nil))

Whichever plugin causing this is really problematic indeed.
